test/chain.test.js

```diff
@@ -25,6 +26,7 @@ const mockPushProcessors = {
   pullRemote: sinon.stub(),
   writePack: sinon.stub(),
   getDiff: sinon.stub(),
+  checkSensitiveData : sinon.stub(),
```

Comment:
By moving this new functionality in its own plugin, it will keep the proxy chain easier to test. Having to constantly add new functions here will not be maintainable long term. This test itself is also not exactly well structured or easy to maintain so we want to not add to it as much as possible. Plugins are preferred.